tensorflow/core/framework/dataset_options.proto

```diff
@@ -139,6 +139,30 @@ enum ExternalStatePolicy {
   POLICY_FAIL = 2;
 }
 
+message AutotuneOptions {
```

Comment:
We should also remove the corresponding fields from the OptimizationOptions and update the C++ to be backwards compatible (it should check whether the old OptimizationOptions fields exist and if so use them and if they don't use the new AutotuneOptions).

Reply:
I have removed the redundant properties from `OptimizationOptions` proto message and retained the encoding code for the remaining properties.

Reply:
Also since these proto message properties have been removed, the respective helper methods will also be removed. Thus, I have made the switch to autotune options directly.